static-assets/components/cstudio-forms/controls/file-name.js

```diff
@@ -164,6 +165,11 @@ YAHOO.extend(CStudioForms.Controls.FileName, CStudioForms.CStudioFormField, {
         el.selectionEnd = cursorPosition;
       }
     }
+
+    if (allowAccent) {
```

Comment:
This looks backwards; when `allowed` then they shouldn't be replaced.
```suggestion
    if (!allowAccent) {
```

Reply:
So if allow, do we want it to be translated to a hyphen? This sounds not correct to me.

Reply:
No, it's not about the hyphens. The line inside the `if` statement is:

```js
el.value = el.value.normalize('NFD').replace(/\p{Diacritic}/gu, '');
```

This seems to remove accents/diacritics and it's being done when they are **allowed**, that's what I find odd.

As I understand it, if the value is `à la carte, pied-à-terre, or crème` and this setting is **on** (allow = true), then the chars should be kept and you should end up with the same string (i.e. `à la carte, pied-à-terre, or crème`, not `a la carte, pied-a-terre, or creme` which is what the normalize.replace produces). 

In code...
```js
'à la carte, pied-à-terre, or crème'.normalize('NFD').replace(/\p{Diacritic}/gu, '') // => a la carte, pied-a-terre, or creme
```

Am I missing something?

Reply:
You are correct about that piece of code alone. 
But after the replacement, we also transform to hyphens for special characters:

```
if (invalid.exec(data) != null) {
      el.value = data.replace(invalid, '-');
      YAHOO.util.Event.stopEvent(evt);
    }
```

So if we don't replace them with English characters, it will eventually be replaced with a hyphen. Do we want to keep accent as a part of the URL? If so, that would be a different requirement.

Reply:
I thought that was part of it. As I understood from Russ, the chars were meaningful and desired even on the URL. Perhaps I misunderstood. That's a question for @russdanner & @sumerjabri.

If the chars aren't kept on the URL, what is it that we are keeping?

Reply:
@rart I think there's confusion about what this does. This enhancement is supposed to change or convert accents/diacritics to Latin characters. So

```
In code...

'à la carte, pied-à-terre, or crème'.normalize('NFD').replace(/\p{Diacritic}/gu, '') // => a la carte, pied-a-terre, or creme
```

is what is expected